### PR TITLE
User Story 89540

### PR DESCRIPTION
**Pull request by TomArcherMsft**
[User Story 89540](https://dev.azure.com/msft-skilling/Content/_workitems/edit/89540)

### CODE REVIEW

**Review by stemaMSFT — CHANGES_REQUESTED** [review]
Hey Tom,

The Azure compute module being used on this is actually for AzureRM 2.X, and isn't recommended for usage in 3.0 onward. I believe @lonegunmanb can confirm that https://github.com/Azure/terraform-azurerm-virtual-machine should be the module used for this implementation. In which case, this may need to be significantly rewritten to support the latest version.

**Comment by TomArcherMsft** [comment]
> Hey Tom,
> 
> The Azure compute module being used on this is actually for AzureRM 2.X, and isn't recommended for usage in 3.0 onward. I believe @lonegunmanb can confirm that https://github.com/Azure/terraform-azurerm-virtual-machine should be the module used for this implementation. In which case, this may need to be significantly rewritten to support the latest version.

@stemaMSFT I understand that the article is out of date with regards to the provider. However, the article as it stands today is broken. Since, it will take several weeks for me to get around to having the time to rewrite this sample code, we have two immediate options: 

- Fix the issues in the current version of the code.
- Remove the article until it's rewritten. 

**I personally believe a functioning older article is better than not having any article. Therefore, my recommendation is to publish the fixes in this PR, and put the rewrite on our backlog.**

File: quickstart/101-vm-cluster-windows/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+output "resource_group_name" {
```
Comment (Jingwei-MS) [inline quickstart/101-vm-cluster-windows/outputs.tf:1]: @TomArcherMsft Shall we include output like host group id and machine scale set id as well?

File: quickstart/101-vm-cluster-windows/main.tf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  prefix = "dns"
+}
+
+module "windowsservers" {
+  source              = "Azure/compute/azurerm"
+  resource_group_name = azurerm_resource_group.rg.name
+  vnet_subnet_id      = module.network.vnet_subnets[0]
+  is_windows_image    = true
+  vm_hostname         = "vm-${random_string.windowsservers_vm_hostname.result}"
+  admin_password      = "ComplxP@ssw0rd!"
```
Comment (lonegunmanb) [inline quickstart/101-vm-cluster-windows/main.tf]: I would recommend using `random_password` to generate a robust password.

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Hi @TomArcherMsft almost LGTM, besides the comment only one question: why this example's name is `vm-cluster-windows`? It looks like we only have one virtual machine in this example.

**Comment by TomArcherMsft** [comment]
> Hi @TomArcherMsft almost LGTM, besides the comment only one question: why this example's name is `vm-cluster-windows`? It looks like we only have one virtual machine in this example.

"Windows" is a trademarked name, so its plurality cannot be changed. If this article were for more than one VM cluster the name could be "vm-clusters-windows" (note the pluralization of the word "cluster"). The current name - "wm-cluster-windows" - is correct.

File: quickstart/101-vm-cluster-windows/outputs.tf (hunk 1 above)
Comment (TomArcherMsft) [reply quickstart/101-vm-cluster-windows/outputs.tf:1]: @Jingwei-MS 

How do I access those two values to output them and what can the customer do with those values? 

To keep the sample simple, we limit the output values to those that are dynamically created and/or values that must be used to complete the tutorial. 

For example:
- If we generate an Azure resource name using something like `random_pet`, we output that value so the customer knows what was created. 
- We also output any values that are required parameters to the Azure CLI and Azure PowerShell commands in the "Verify the results" section of the article.

File: quickstart/101-vm-cluster-windows/main.tf (hunk 2 above)
Comment (TomArcherMsft) [reply quickstart/101-vm-cluster-windows/main.tf]: Done

**Comment by TomArcherMsft** [comment]
@lonegunmanb Closing PR. Will open as new PR.

**Pull request closed without merging**